src/tests/e2etest/CommandRunnerTests.cs

```diff
@@ -1,19 +1,279 @@
+            public int ac { get; set; }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _registryManager = RegistryManager.CreateFromConnectionString(_iotHubConnectionString);
+
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Assert.Warn("Missing environment vars required for log upload to blob store");
+            }
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Console.WriteLine("[TearDown] Uploading logs to blob store");
+
+                string fileName = String.Format("{0}-{1}.tar.gz", _resourceGroupName, _deviceId);
+                string fullURL = String.Format("{0}{1}?{2}", _uploadUrl, fileName, _sasToken);
+
+                string tempFileCommand = "temp_file=`sudo mktemp`";
+                string tarCommand = "sudo tar -cvzf $temp_file osconfig*.log";
+                string curlCommand = String.Format("curl -X PUT -T $temp_file -H \"x-ms-date: $(date -u)\" -H \"x-ms-blob-type: BlockBlob\" \"{1}\"", fileName, fullURL);
+                string uploadCommand = String.Format("cd /var/log && {0} && {1} && {2}", tempFileCommand, tarCommand, curlCommand);
+
+                // Temporarily commented out to avoid uploading logs to blob store twice
+                // if (!ExecuteCommandViaCommandRunner(uploadCommand))
+                // {
+                //     Assert.Warn("[TearDown] Failed to upload logs to blob storage");
+                // }
+            }
+            else
+            {
+                Console.WriteLine("[TearDown] Skipping upload of logs to blob store");
+            }
+        }
+
+        protected bool ExecuteCommandViaCommandRunner(string arguments)
+        {
+            bool success = true;
+            var command = CommandRunnerTests.CreateCommand(arguments);
+
+            try
+            {
+                var desiredResult = SetDesired<CommandRunnerTests.CommandArguments>("CommandRunner", "CommandArguments", command);
+                desiredResult.Wait();
+                int ackCode = desiredResult.Result;
+
+                if (200 != ackCode)
+                {
+                    Console.WriteLine("[ExecuteCommandViaCommandRunner] Failed to set desired state");
+                    success = false;
+                }
+                else
+                {
+                    Func<CommandRunnerTests.CommandStatus, bool> condition = (CommandRunnerTests.CommandStatus status) =>
+                    {
+                        return status.CommandId == command.CommandId && status.CurrentState == CommandRunnerTests.CommandState.Succeeded;
+                    };
+
+                    var reportedResult = GetReported<CommandRunnerTests.CommandStatus>("CommandRunner", "CommandStatus", condition, 180);
```

Comment:
what does the `180` represent?

Reply:
double of your 90 timeout im guessing? Can we find a constant to use based on others?

Reply:
Good call! I'll base this number (and others) off of the env variable previously used for timeout.